entity-service/src/test/kotlin/com/egm/stellio/entity/service/EntityServiceTests.kt

```diff
@@ -826,6 +826,131 @@ class EntityServiceTests {
         confirmVerified()
     }
 
+    @Test
+    fun `it should create a new multi attribute property`() {
+
+        val entityId = "urn:ngsi-ld:Beehive:123456"
+        val newProperty = """
+            {
+              "fishNumber": [{
+                "type": "Property",
+                "value": 500,
+                "datasetId": "urn:ngsi-ld:Dataset:fishNumber:1"
+              },
+              {
+                "type": "Property",
+                "value": 600
+              }]
+            }
+        """.trimIndent()
+        val expandedNewProperty =
+            NgsiLdParsingUtils.expandJsonLdFragment(
+                newProperty,
+                aquacContext!!
+            )
+
+        val mockkedEntity = mockkClass(Entity::class)
+
+        every { mockkedEntity.id } returns entityId
+        every { mockkedEntity.properties } returns mutableListOf()
+
+        every { neo4jRepository.hasPropertyInstance(any(), any()) } returns false
+        every { entityRepository.findById(any()) } returns Optional.of(mockkedEntity)
+        every { mockkedEntity.id } returns entityId
+        every { neo4jRepository.createPropertyOfSubject(any(), any()) } returns UUID.randomUUID().toString()
+
+        entityService.appendEntityAttributes(entityId, expandedNewProperty, false)
+
+        verify {
+            neo4jRepository.hasPropertyInstance(
```

Comment:
A suggestion to avoid duplicating the almost identical `hasPropertyInstance` calls:

1. Capture the datasetId parameter

```
val datasetSetIds = mutableListOf<URI>()
every { neo4jRepository.hasPropertyInstance(any(), any(), capture(datasetSetIds)) } returns false
```

2. Check the list only contains one element

`assertTrue(datasetSetIds.contains(URI.create("urn:ngsi-ld:Dataset:fishNumber:1"))) `

3. Only do one verify call

```
verify(exactly = 2) {
       neo4jRepository.hasPropertyInstance(
            match {
                it.id == entityId &&
                    it.label == "Entity"
            },
            "https://ontology.eglobalmark.com/aquac#fishNumber",
            datasetId = any()
        )
    }
```